devices/RgbDiode/RgbDiode.cs

```diff
@@ -72,6 +80,92 @@ public void SetColor(Color color)
             SetValue(_rChannel, color.R * _rFactor);
             SetValue(_gChannel, color.G * _gFactor);
             SetValue(_bChannel, color.B * _bFactor);
+            CurrentColor = color;
+        }
+        
+        /// <summary>
+        /// Transitions the pixel to new color in a separate thread.
+        /// </summary>
+        /// <param name="color">The new color.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <param name="steps">Number of transition steps.</param>
+        /// <param name="delay">Delay between each transition step.</param>
+        /// <returns>The new Thread or null if color is the same as current color.</returns>
+        public Thread TransitionAsync(Color color, CancellationToken cancellationToken, int steps = DefaultSteps, int delay = DefaultDelay)
```

Comment:
why do you return the thread? Would it be better to return a cancellation token or equivalent rather than the thread itself?
Also any sample on how to properly use this new feature? :-)

Reply:
Well, I've implemented it based on "Task" approach from full .NET (maybe that's wrong approach for Threads?). But, in my opinion token is something that we are providing from outside of method - to be able to cancel thread. 
I think, I should add default to CancellationToken.

As for sample, I'll provide one in readme and sample soon.

Reply:
Included sample